### DISCUSSION TITLE
Transmission 2.94 not using UDP upnp

### DISCUSSION THREAD

**Discussion by axet** [transmission/transmission#8488] [Support]
Hello!

I'm using transmission 2.94 with my openwrt router with miniupnpd. By looking at 'cat /tmp/upnp.leases' I can see only:

TCP:51413:192.168.54.178:51413:1589268679:NAT-PMP 51413 tcp

But no UDP connection mapping. And it seems no one can connect using UDP to my torrents.

My router does support UDP like this (with other software):

UDP:45615:192.168.54.100:45615:1589265570:-GT0002-

Technically speaking TCP/IP and TCP/UDP are the same TCP family protocol, but I'm not sure that how it works with miniupnpd.

Another issue I have with port checking. Since I have two IPv4 and IPv6, port checking using default routing which is IPv6 and result port checking always failed. Can I specify which connection to use to test port? or can you support double connection for port checking (IPv4 and IPv6)?

Thanks very much!

EDIT: I can confirm sharing torrents with qBittorrent works fine!

TCP:8999:192.168.54.178:8999:1589870795:qBittorrent/4.1.5 at 192.168.54.178:8999
UDP:8999:192.168.54.178:8999:1589870795:qBittorrent/4.1.5 at 192.168.54.178:8999

**Comment by lnpz1** [comment]
I think this could be more related to NAT-PMP than UPnP. I also use OpenWrt with miniupnpd. When UPnP is disabled, NAT-PMP is enabled, Transmission 2.94 (d8e60ee) only creates TCP mapping, when both NAT-PMP and UPnP are enabled, it sometimes creates both TCP and UDP mappings, sometimes only TCP mapping, so I think it depends on which service is discovered first.

There's also another problem with UPnP in a dual stack network which I reported here https://github.com/transmission/transmission/issues/1243
